File: java/pl/edu/pg/mygame/organisms/animals/Fox.java
package pl.edu.pg.mygame.organisms.animals;

import pl.edu.pg.mygame.Grid;
import pl.edu.pg.mygame.organisms.Species;

import java.awt.*;
import java.util.ArrayList;
import java.util.List;

import static pl.edu.pg.mygame.utils.HelperFunctions.getRandomPoint;

public class Fox extends Animal {

    public Fox(Point p) {
        super(p);
        this.species = Species.FOX;
        this.initiative = 7;
        this.strength = 3;
    }

    @Override
    protected Point getNewPoint(Grid grid) {
        List<Point> neighbours = grid.getNeighbourCells(position);
        List<Point> final_neighbours = new ArrayList<>();
        for (Point neighbour : neighbours) {
            if ((grid.getOrganism(neighbour) != null && grid.getOrganism(neighbour).getStrength() <= this.strength) ||
                     (grid.getOrganism(neighbour) == null))
                final_neighbours.add(neighbour);
        }
        if (final_neighbours.size() > 0)
            return getRandomPoint(final_neighbours);
        return null;
    }
}

File: python/organisms/animals/fox.py
from organisms.animals.animal import Animal
from organisms.species import Species
from utils.helpers import Helpers


class Fox(Animal):
    def __init__(self, pos):
        super().__init__(pos)
        self.strength = 3
        self.initiative = 7
        self.species = Species.FOX

    def get_new_point(self, grid):
        neighbors = grid.get_neighbor_cells(self.position)
        final_neighbors = []
        for neighbor in neighbors:
            if (grid.get_organism(neighbor) is not None and grid.get_organism(neighbor).strength <= self.strength) or (grid.get_organism(neighbor) is None):
                final_neighbors.append(neighbor)
        if len(final_neighbors) > 0:
            return Helpers.get_random_point(final_neighbors)
        return None
